chatkit/src/main/java/com/pusher/chatkit/CurrentUser.kt

```diff
@@ -173,34 +183,114 @@ class CurrentUser(
         }
     }
 
-    fun addMessage(
-            text: String,
-            room: Room,
+    fun sendMessage(
+            roomId: Int,
+            text: String? = null,
+            attachment: GenericAttachment? = null,
             onCompleteListener: MessageSentListener,
             onErrorListener: ErrorListener
-    ){
-        val messageReq = MessageRequest(
+    ): Cancelable? {
+        var attachmentBody: AttachmentBody? = null
+
+        if (attachment != null) {
```

Comment:
We can use some async await magic here if you want to clean it up. Not that I am against this, but I reverted to this just to be able to debug that error we had.